index.bs

```diff
@@ -691,21 +691,124 @@ An {{MLOperand}} represents an intermediary graph being constructed as a result
+1. Let |builder| be the first argument.
+1. If |builder| is not an instance of {{MLGraphBuilder}}, throw a "{{TypeError}}" and abort these steps.
+1. Set [=this=]'s {{MLOperand/[[builder]]}} to |builder|.
+1. Let |kind| be the second argument.
+1. If |kind| is not an {{MLOperandKind}}, throw a "{{TypeError}}" {{DOMException}} and abort these steps.
+1. Set [=this=]'s {{MLOperand/[[kind]]}} to |kind|.
+1. Let |type| be the third argument.
+1. If |type| is not an {{MLOperandType}}, throw a "{{TypeError}}" {{DOMException}} and abort these steps.
+1. Set [=this=]'s {{MLOperand/[[type]]}} to |type|.
+1. If |kind| is `"constant"`:
+    1. Let |name| be the fourth argument.
+    1. If |name| is not a [=string=], throw a "{{TypeError}}" {{DOMException}} and abort these steps.
+    1. Set [=this=]'s {{MLOperand/[[name]]}} to |name|.
+    1. Let |desc| be the fifth argument.
+    1. If |desc| is not an {{MLOperandDescriptor}}, throw a "{{TypeError}}" {{DOMException}} and abort these steps.
+    1. If |type| is not equal to |desc|'s {{MLOperandDescriptor/type}}, throw a "{{DataError}}" {{DOMException}} and abort these steps.
+    1. Set [=this=]'s {{MLOperand/[[dimensions]]}} to |desc|.{{MLOperandDescriptor/dimensions}}.
+    1. Let |bufferView| be the sixth argument.
+    1. If |bufferView| is not an {{MLBufferView}}, throw a "{{DataError}}" {{DOMException}} and abort these steps.
+    1. If |type| is not equal to |bufferView|'s [=element type=], throw a "{{DataError}}" {{DOMException}} and abort these steps.
+    1. If the [=byte length=] of |desc| is not supported by the implementation, throw a "{{DataError}}" {{DOMException}} and abort these steps.
+    1. If the [=byte length=] of |desc| is not equal with |bufferView|.\[[ByteLength]], throw a "{{DataError}}" {{DOMException}} and abort these steps.
+    1. Set [=this=]'s {{MLOperand/[[tensorData]]}} to |bufferView|.
+1. Otherwise, of |kind| is `"input"`:
+    1. Let |name| be the fourth argument.
+    1. If |name| is not a [=string=], throw a "{{TypeError}}" {{DOMException}} and abort these steps.
+    1. Set [=this=]'s {{MLOperand/[[name]]}} to |name|.
+    1. Let |desc| be the fifth argument.
+    1. If |desc| is not an {{MLOperandDescriptor}}, throw a "{{TypeError}}" {{DOMException}} and abort these steps.
+    1. If |type| is not equal to |desc|'s {{MLOperandDescriptor/type}}, throw a "{{DataError}}" {{DOMException}} and abort these steps.
+    1. If the [=byte length=] of |desc| is not supported by the implementation, throw a "{{DataError}}" {{DOMException}} and abort these steps.
+    1. Set [=this=]'s {{MLOperand/[[dimensions]]}} to |desc|.{{MLOperandDescriptor/dimensions}}.
+1. Otherwise, of |kind| is `"output"`:
+    1. Let |dimensions| be the fourth argument.
+    1. If the [=check dimensions=] steps given |type| and |dimensions| return `false`, throw a "{{DataError}}" {{DOMException}} and abort these steps.
+    1. Let |activation| be the fifth argument.
```

Comment:
I don't think `|activation|` should be used here.  `|activation|` is only used as activation function that may be fused into an operator. Instead, could the platform object representing an operator be used? The operator produces this MLOperand.

Reply:
I agree, but it would be nice to have a definition for an operator. An obscure internal object exposed in an internal slot doesn't sound like a clear definition. :) 
I wonder if defining MLOperator and also keeping MLActivation would work better? since now we have only MLActivation defined in the spec but MLOperator is used in the implementation.

In that sense, MLOperator would be similarly empty object like MLGraph currently is. Of course Occam tells that if we can do without MLOperator in the spec, we should. But then we need a clear "private" definition of an operator object (for internal use).